src/mesh.f90

```diff
@@ -0,0 +1,325 @@
+module m_mesh
+  use iso_fortran_env, only: stderr => error_unit
+
+  use mpi
+  use m_common, only: dp, DIR_X, DIR_Y, DIR_Z, DIR_C, &
+                     CELL, VERT, NONE, X_FACE, Y_FACE, Z_FACE, &
+                     X_EDGE, Y_EDGE, Z_EDGE
+  use m_field, only: field_t
+
+  implicit none
+
+
+  ! Stores geometry information
+  type :: geo_t
+    real(dp), dimension(3) :: d ! size of a cell in each direction (=edge length, distance between centers, distance between vertices)
+    real(dp), dimension(3) :: L ! Global dimensions of the domain in each direction
+  end type
+
+  ! Stores parallel domain related information
+  type :: parallel_t
+    integer :: nrank ! local rank ID
+    integer :: nproc ! total number of ranks/proc participating in the domain decomposition
+    integer, dimension(3) :: nrank_dir ! local rank ID in each direction
+    integer, dimension(3) :: nproc_dir ! total number of proc in each direction
+    integer, dimension(3) :: n_offset  ! number of cells offset in each direction due to domain decomposition
+    integer, dimension(3) :: pnext ! rank ID of the previous rank in each direction
+    integer, dimension(3) :: pprev ! rank ID of the next rank in each direction
```

Comment:
There are few members here each with lenght 3. One alternative is having a struct with members `nrank_dir`, `nproc_dir`, etc. and have this new struct in a lenght 3 array. 
We can also discuss about naming the spatial dimensions. Mesh object uses lenght 3 arrays but we access the `dirps` objects with `x/y/zdirps`. If people prefer numbering we can change the naming for `dirps` objects too, and maybe have them in an array with size 3 instead.

Reply:
I'd rather keep it this way as you may sometimes need to know things in multiple directions and it is consistent with the usage of other `mesh` fields.